### PR TITLE
gh-2865 Cache Exceptions usage and review

### LINKED ISSUE #2865: Cache Exceptions usage and review
**Describe the bug**
Following #2319 The caches now share a parent class `Cache`. However this combined the need for certain Exceptions to be thrown and required by some test. These Exceptions need to be reviewed. One example is a test to for when a Key is not found in the Cache an Exception is thrown. Keys not being found in a cache is not "exceptional behaviour".

### PR DESCRIPTION
**Pull request by GCHQDev404**
# Related Issue

- Resolve #2865

### CODE REVIEW

File: core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/named/cache/NamedOperationCache.java [hunk 1]
```diff
@@ -145,36 +138,23 @@ public NamedOperationDetail getFromCache(final String name) throws CacheOperatio
...
+    public void addNamedOperation(final NamedOperationDetail namedOperation, final boolean overwrite, final User user, final String adminAuth) throws CacheOperationException {
+        if (namedOperation == null) {
+            throw new CacheOperationException("NamedOperation cannot be null");
         }
-        if (!overwrite) {
-            addToCache(name, namedOperation, false);
-            return;
+        String name = namedOperation.getOperationName();
+        if (contains(name) && overwrite) {
+            final boolean doesUserHavePermissionToWrite = getFromCache(name).hasWriteAccess(user, adminAuth);
```
Comment (t92549) [inline core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/named/cache/NamedOperationCache.java:147]: Shouldn't user need write access to perform normal add also (not overwriting)?

Comment (GCHQDev404) [reply core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/named/cache/NamedOperationCache.java:147]: So in this situation.

I add a NamedOperation to cache, but I specify you have write permissions. This means you can override my NamedOperation. 

Else where in Gaffer land there will be checks for if you can use the AddNamedOperation, But here is where a 3rd person is stopped from overriding my namedOperation, but who has AddNamedOperation permission.

Comment (t92549) [reply core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/named/cache/NamedOperationCache.java:147]: Is it possible to add a NamedOperation that I will not be able to access later?

Comment (t92549) [reply core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/named/cache/NamedOperationCache.java:147]: This shouldn't happen and if it does it is an unrelated preexisting issue

**Review by t92549 — APPROVED** [review]

**Pull request merged**